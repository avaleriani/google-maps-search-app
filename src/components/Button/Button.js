import React from "react";

import styles from "./styles.module.css";

export default ({ label, ...props }) => (
  <button className={styles["button"]} {...props}>
    {label}
  </button>
);
