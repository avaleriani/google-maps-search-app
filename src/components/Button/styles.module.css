.button {
  padding: 10px 30px;
  font-size: 15px;
  border: none;
  background: #00ead4;
  color: #fff;
  border-radius: 3px;
  transition: all 0.3s ease;
  cursor: pointer;
  outline: none;
}

.button:hover {
  background: #15E2CF;
  box-shadow: 0px 4px 19px -2px rgba(0, 255, 231, 0.53);
}