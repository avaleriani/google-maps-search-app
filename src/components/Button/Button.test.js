import React from 'react';
import Button from './Button';
import renderer from 'react-test-renderer';


it('renders correctly', () => {
  const component = renderer.create(
    <Button/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('has label', () => {
  const label = "test label"
  const component = renderer.create(
    <Button label={label}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});