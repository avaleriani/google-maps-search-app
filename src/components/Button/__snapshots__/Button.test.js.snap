// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has label 1`] = `
<button
  className="button"
>
  test label
</button>
`;

exports[`renders correctly 1`] = `
<button
  className="button"
/>
`;
